raindrops/script.js

```diff
@@ -0,0 +1,584 @@
+    return lastFirstOperand;
+}
+
+function setSecondOperand() {
+    lastSecondOperand = Math.round(secondOperandMin - 0.5 + Math.random() * (secondOperandMax - secondOperandMin + 1));
+    if (lastOperator == '/' && lastFirstOperand % lastSecondOperand !== 0) {
+        console.log("Замена второго операнда: не / без остатка");
+        for (let i = Math.ceil(secondOperandMin); i <= Math.floor(secondOperandMax); i++) {
+            if (lastFirstOperand % i === 0) {
+                setSecondOperand();
+                break;
+            } else {
+                lastSecondOperand = lastFirstOperand;
+            }
+        }      
+    }
+    if (lastOperator == '-' && lastFirstOperand <= lastSecondOperand) {
+        console.log("Замена второго операнда: отрицательное или 0 значение при -");
+        if (lastFirstOperand <= Math.ceil(secondOperandMin)) {
+            lastSecondOperand = lastFirstOperand - 1;
+        } else {
+            setSecondOperand();
+        }
+    }
+    if (lastOperator == '*' && !manualSetOperand) {
+        lastSecondOperand = Math.round(lastSecondOperand / 2);    
+    }
+    return lastSecondOperand;
+}
+
+function pressCalcBtn(value) {
+    if (value === 'Clear') {
+        displayOutput.value = '';
+    } else if (value === 'Delete') {
+        displayOutput.value = displayOutput.value.slice(0, -1);
+    } else if (value === 'Enter') {
+        checkAnswer(displayOutput.value);
+        displayOutput.value = '';
+    } else {
+        if (+displayOutput.value > 999) {
+            displayOutput.value = displayOutput.value;
+        } else if (displayOutput.value === '' && value === '0') {
+            displayOutput.value = '';
+        } else {
+            displayOutput.value += value;
+        }
+    }
+};
+
+function pressKeyboard(e) {
+    if (e.code.includes('Digit') || e.code.includes('Numpad') && (e.keyCode >= 96 && e.keyCode <= 105)) {
+        if (+displayOutput.value > 999) {
+            displayOutput.value = displayOutput.value;
+        } else if (displayOutput.value === '' && e.code.includes('0')) {
+            displayOutput.value = '';
+        } else {
+            displayOutput.value = displayOutput.value + e.key;
+        }
+    };
+    if (e.code.includes('Enter')) {
+        checkAnswer(displayOutput.value);
+        displayOutput.value = '';
+    }
+    if (e.code.includes('Backspace')) {
+        displayOutput.value = displayOutput.value.slice(0, -1);
+    }
+}
+
+function checkAnswer() {
+    if (rainDropsArr[0].result === Number(displayOutput.value)) {
+        playCorrectAnswerSound();
+        getGameScore('plus');
+        ++counterRightAnswers;
+        rainDropsArr[0].wasDecided = true;
+        rainDropsArr[0].element.style.transform = 'scale(0)';
+        rainDropsArr[0].element.style.transition = 'transform 1s';
+        if (rainDropsArr[0].rainDropExtra === true) {
+            rainDropsArr[0].element.style.boxShadow = '0 0 5px 10px rgba(187, 255, 0, 0.8)';
```

Comment:
If it's possible better way to add styles use CSS classes for it and manage add/remove such classes